src/mpid/ch4/src/ch4_init.c

```diff
@@ -457,6 +457,17 @@ int MPID_Init(void)
     mpi_errno = MPIDU_Init_shm_init();
     MPIR_ERR_CHECK(mpi_errno);
 
+    /* Initialize multiple VCIs */
+    MPL_COMPILE_TIME_ASSERT(sizeof(MPIDI_vci_t) == MPL_CACHELINE_SIZE);
```

Comment:
If there is compilers that doesn't support the alignment attribute, we would need a different solution. It is critical to ensure the size of `MPIDI_vci_t` be greater than cacheline size, to prevent false sharing.

Reply:
Of the ones we test in Jenkins, all appear to support the alignment attribute. However `suncc` differs in the way `sizeof` is affected. Test program

```
#include <stdio.h>

int main(void) {
  struct foo {
    int a;
  } __attribute__((aligned(64)));

  printf("sizeof(struct foo) = %zu\n", sizeof(struct foo));
  printf("alignof(struct foo) = %zu\n", _Alignof(struct foo));

  return 0;
}
```

And output with `suncc -std=c11`

```
sizeof(struct foo) = 4
alignof(struct foo) = 64
```

Reply:
Just added a "TODO" comment for now.